MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,12 +1,175 @@
+             case runtimeMinsFailure
+         }
+    
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        alertPresenter = AlertPresenter()
+        alertPresenter?.delegate = self
+        questionFactory = QuestionFactory(delegate: self)
+        questionFactory?.requestNextQuestion()
+        statisticService = StatisticServiceImplementation()
+        
+        let documentDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
+        let fileURL = documentDirectory.appendingPathComponent("top250MoviesIMDB.json")
+        var jsonString = ""
+
+        do {
+            jsonString = try String(contentsOf: fileURL)
+        } catch FileManagerError.fileDoesntExist {
+            print("Файл по адресу \(fileURL) не существует")
+        } catch {
+            print("Неизвестная ошибка чтения из файла \(fileURL)")
+        }
+
+        let data = jsonString.data(using: .utf8)!
+
+        do {
+            let result = try JSONDecoder().decode(Top.self, from: data)
+        } catch {
+            print("Failed to parse \(jsonString)")
+        }
+    }
+    
+    // MARK: - QuestionFactoryDelegate
+
+    func didReceiveNextQuestion(question: QuizQuestion?) {
+        guard let question = question else {
+            return
+        }
+        
+        currentQuestion = question
+        let viewModel = convert(model: question)
+        DispatchQueue.main.async { [weak self] in
+            self?.show(quiz: viewModel)
+        }
+    }
+    
+    // MARK: - AlertPresenterDelegate
+    
+    func didPresentAlert(alert: UIAlertController?) {
+        guard let alert = alert else {
+            return
+        }
+        DispatchQueue.main.async { [weak self] in
+            self?.present(alert, animated: true, completion: nil)
+        }
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        
+        if isCorrect {
+            correctAnswers += 1
+        }
+        
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.cornerRadius = 20
+        imageView.layer.borderColor = isCorrect ? UIColor(named: "YP Green")?.cgColor : UIColor(named: "YP Red")?.cgColor
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
+            self?.showNextQuestionOrResults()
+            
+        }
+    }
+    
+    private func show(quiz step: QuizStepViewModel) {
+        countLabel.text = step.questionNumber
+        imageView.image = step.image
+        questionLabel.text = step.question
+    }
+    
+    private func showNextQuestionOrResults() {
+        
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 0
+        
+        if currentQuestionIndex == questionsAmount - 1 {
+        statisticService?.store(correct: correctAnswers, total: questionsAmount)
+        guard let gamesCount = statisticService?.gamesCount else {return}
```

Comment:
Здесь будет выгоднее разыменовать statisticService. Так до свойств можно будет добраться без разыменовывания, а здесь пробелы добавить:
```suggestion
        guard let gamesCount = statisticService?.gamesCount else { return }
```